schema-engine/connectors/schema-connector/src/warnings.rs

```diff
@@ -0,0 +1,611 @@
+    pub remapped_enums: Vec<Enum>,
+    /// The relation is copied from a previous data model, only if
+    /// `relationMode` is `prisma`.
+    pub reintrospected_relations: Vec<Model>,
+    /// The name of these models or enums was a dupe in the PSL.
+    pub duplicate_names: Vec<TopLevelItem>,
+    /// Warn about using partition tables, which only have introspection support.
+    pub partition_tables: Vec<Model>,
+    /// Warn about using inherited tables, which only have introspection support.
+    pub inherited_tables: Vec<Model>,
+    /// Warn about non-default NULLS FIRST/NULLS LAST in indices.
+    pub non_default_index_null_sort_order: Vec<IndexedColumn>,
+    /// Warn about using row level security, which is currently unsupported.
+    pub row_level_security_tables: Vec<Model>,
+    /// Warn about check constraints.
+    pub check_constraints: Vec<ModelAndConstraint>,
+    /// Warn about exclusion constraints.
+    pub exclusion_constraints: Vec<ModelAndConstraint>,
+    /// Warn about row level TTL
+    pub row_level_ttl: Vec<Model>,
+    /// Warn about non-default unique deferring setup
+    pub non_default_deferring: Vec<ModelAndConstraint>,
+    /// Warn about comments
+    pub objects_with_comments: Vec<Object>,
+    /// Warn about fields which point to an empty type.
+    pub model_fields_pointing_to_an_empty_type: Vec<ModelAndField>,
+    /// Warn about compositefields which point to an empty type.
+    pub type_fields_pointing_to_an_empty_type: Vec<TypeAndField>,
+    /// Warn about unknown types in a model.
+    pub model_fields_with_unknown_type: Vec<ModelAndField>,
+    /// Warn about unknown types in a composite type.
+    pub type_fields_with_unknown_type: Vec<TypeAndField>,
+    /// Warn about undecided types in a model.
+    pub undecided_types_in_models: Vec<ModelAndFieldAndType>,
+    /// Warn about undecided types in a composite type.
+    pub undecided_types_in_types: Vec<TypeAndFieldAndType>,
+}
+
+impl Warnings {
+    /// Generate a new empty warnings structure.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// True if we have no warnings
+    pub fn is_empty(&self) -> bool {
+        self == &Self::default()
+    }
+
+    /// True, if the datamodel has Prisma 1 style defaults
+    pub fn uses_prisma_1_defaults(&self) -> bool {
+        !self.prisma_1_uuid_defaults.is_empty() || !self.prisma_1_cuid_defaults.is_empty()
+    }
+}
+
+impl fmt::Display for Warnings {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str("*** WARNING ***\n")?;
+
+        fn render_warnings<T>(msg: &str, items: &[T], f: &mut fmt::Formatter<'_>) -> fmt::Result
+        where
+            T: fmt::Display,
+        {
+            if !items.is_empty() {
+                writeln!(f)?;
+                f.write_str(msg)?;
+                writeln!(f)?;
+
+                for item in items {
+                    writeln!(f, "  - {item}")?;
```

Comment:
Do we want the extra spacing here before `- ...` it looks a little weird to me combined with the newline